ui/app/models/totp.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Model, { attr } from '@ember-data/model';
+import { alias } from '@ember/object/computed';
+import { computed } from '@ember/object';
+import { withModelValidations } from 'vault/decorators/model-validations';
+import fieldToAttrs, { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+
+const ALGORITHMS = ['SHA1', 'SHA256', 'SHA512'];
+const DIGITS = [6, 8];
+
+const validations = {
+  account_name: [
+    { type: 'presence', message: "Account name can't be blank." },
+    {
+      type: 'containsWhiteSpace',
+      message:
+        "Account name contains whitespace. If this is desired, you'll need to encode it with %20 in API requests.",
+      level: 'warn',
+    },
+  ],
+  algorithm: [
+    {
+      validator: (value) => ALGORITHMS.includes(value),
+      message: 'Algorithm must be one of ' + ALGORITHMS.join(', '),
+    },
+  ],
+  digits: [
+    { validator: (value) => DIGITS.includes(value), message: 'Digits must be one of ' + DIGITS.join(', ') },
+  ],
+  period: [{ type: 'number', message: 'Period must be a number.' }],
+};
+
+@withModelValidations(validations)
+export default class TotpModel extends Model {
+  @attr('string') backend;
+
+  @alias('account_name') name;
+
+  @attr('string', {
+    fieldValue: 'name',
+    readOnly: true,
+  })
+  account_name;
+  @attr('string', {
+    readOnly: true,
+    possibleValues: ALGORITHMS,
+  })
+  algorithm;
+  @attr('number', {
+    readOnly: true,
+    possibleValues: DIGITS,
+  })
+  digits;
+  @attr('string', {
+    readOnly: true,
+  })
+  issuer;
+  @attr('number', {
+    readOnly: true,
+  })
+  period;
+
+  @attr('string') url;
+
+  @computed('account_name', function () {
```

Comment:
I'm not sure about this computed property. Now the `account_name` form field loses focus on every keystroke. Does `attrs` need to be a computed property? Or could it just be a getter?

Reply:
Right, this should be a getter! In fact all of the `@computed` values should be a getter instead.